ROX-18395: compliance check results api

dashrews78 opened the pull request:
## Description

protos for v2 compliance operator results apis

## Checklist
- [ ] Investigated and inspected CI test results
~- [ ] Unit test and regression tests added~
~- [ ] Evaluated and added CHANGELOG entry if required~
~- [ ] Determined and documented upgrade steps~
~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs]~(https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed
 Just protos.  CI sufficient.

dashrews78 committed:
X-Smart-Branch-Parent: master

md2119 commented on proto/api/v2/compliance_results_service.proto:
```diff
@@ -0,0 +1,142 @@
+}
+
+// ComplianceScanResultsOverview provides an overview and stats of the results of a scan configuration
+message ComplianceScanResultsOverview {
+    ComplianceScanStatsShim       scan_stats            = 1;
+    repeated string               profile_name          = 2;
+    repeated string               cluster_id            = 3;
+}
```
I made a guess and concluded that this object would serve the ring pie charts. This object may make sense for the UI but not so much from API-only standpoint. How about we represent profiles and clusters as passing and failing where passing means all checks must have passed?

md2119 replied:
Fair enough. We can expand the response to indicate pass/fail if it becomes desirable.

md2119 reviewed (approved):
Address the comments before merging.

dashrews78 commented on proto/storage/compliance_operator_v2.proto:
```diff
@@ -96,50 +106,50 @@ message ComplianceOperatorScanSettingV2 {
 // Next Tag: 13
 // Scan object per cluster
 message ComplianceOperatorScanV2 {
-    string              id          = 1 [(gogoproto.moretags) = 'sql:"pk"'];
-    string              scan_name        = 2 [(gogoproto.moretags) = 'search:"Compliance Scan Name,hidden,store" sql:"fk(ComplianceOperatorScanSettingV2:scan_name),no-fk-constraint,index=category:unique;name:scan_unique_indicator"'];
-    string                    cluster_id = 3 [(gogoproto.moretags) = 'search:"Cluster ID,hidden,store" sql:"fk(Cluster:id),no-fk-constraint,type(uuid),index=category:unique;name:scan_unique_indicator"'];
-    repeated string           errors = 4;
-    repeated ProfileShim      profile                 = 5;
-    map<string, string> labels      = 6;
-    map<string, string> annotations = 7;
-    ScanType            scan_type   = 8;
-    NodeRole            node_selector = 9;
-    ScanStatus          status = 10;
-    google.protobuf.Timestamp     created_time             = 11;
-    google.protobuf.Timestamp     last_executed_time        = 12;
```
Made `RuleSeverity` an Enum and then formatting while I was in the file.  I'm waffling on whether or not to keep `RuleSeverity` in the storage proto.  If we are going to eventually use it, it will save migration hassle.  If we never use it we will have a dead column.  My inclination is to probably keep it in storage.  Thoughts?

dashrews78 committed:
X-Smart-Squash: Squashed 7 commits:

b8ba946e05 first cut
c2916c0107 some more work
0a991ea657 some more progress
bb34f2aff9 some groupings and an enum
cd4f221222 various updates
33a7b75f92 cleanup and formatting
db2945659d some feedback work

dashrews78 committed:
rebasing
